pkg/install/installer.go

```diff
@@ -110,6 +114,17 @@ func (i *Installer) collectArtifacts(installation profilesv1.ProfileInstallation
 	return artifacts, nil
 }
 
+func validateProfileArtifact(p *profilesv1.Profile) error {
+	if p.Source.Tag != "" && p.Source.Branch != "" {
+		return fmt.Errorf("cannot configure both %q and %q in profile artifact", "profile.Source.Tag", "Profile.Source.Branch")
+	}
+
+	if p.Source.Tag == "" && p.Source.Branch == "" {
+		return fmt.Errorf("one of %q or %q must be configured", "profile.Source.Tag", "Profile.Source.Branch")
+	}
```

Comment:
these error formats are a bit odd, but wanted to be clear to the person writing the yaml what fields aren't set correctly.